IdentityCore/src/cache/MSIDToken.m

```diff
@@ -295,6 +319,12 @@ - (void)fillFromRequest:(MSIDRequestParameters *)requestParams
 {
     _authority = requestParams.authority;
     _clientId = requestParams.clientId;
+    
+    if ([requestParams isKindOfClass:[MSIDAADV1RequestParameters class]])
+    {
+        MSIDAADV1RequestParameters *v1RequestParams = (MSIDAADV1RequestParameters *)requestParams;
+        _resource = v1RequestParams.resource;
+    }
 }
```

Comment:
We'd need to do the same for the scopes.

Reply:
Nope, the difference here is in MSAL we always save scopes that we're getting back in response. Whereas in ADAL there might be cases where we want to save resource provided by the developer, so we shouldn't save scopes from the request parameters.